contracts/Bond.sol

```diff
@@ -94,6 +94,13 @@ contract Bond is
         uint256 _convertibleRatio,
         uint256 maxSupply
     ) external initializer {
+        /*
+            Safety check: Ensure multiplication can not overflow uint256.
+        */
+        maxSupply * maxSupply;
+        maxSupply * _collateralRatio;
+        maxSupply * _convertibleRatio;
```

Comment:
The `maxSupply * _convertibleRatio;` is probably unnecessary since we have another check that  _convertibleRatio <= _collateralRatio in bondFactory... 

Gas cost is negligible.... let's leave it here for now to make sure all the multiplication cases are covered. In the future let's explore moving all the bond invariants into the initializer function instead of `createBond`